lib/ValidateDirectiveVisitor.ts

```diff
@@ -603,6 +654,20 @@ export const wrapFieldResolverResult = <TContext>(
   };
 };
 
+export const setFieldResolveToValidateAndThenApplyOriginalResolve = <TContext>(
+  field: GraphQLField<unknown, TContext>,
+  validate: ValidateFunction<TContext>,
+  objectType: GraphQLObjectType,
+): void => {
+  const { resolve = defaultFieldResolver, type } = field;
+  // eslint-disable-next-line no-param-reassign
+  field.resolve = async function (...args): Promise<unknown> {
+    validate(undefined, type, objectType, args[2], args[3], args[0], args[1]);
```

Comment:
dont we need the original value in here?

Reply:
In this function we apply the validateFunction before calling the original resolver, therefore no value was generated yet.

This function is called for output (or 'normal') types if the directive implementation overloads the  `applyValidationToOutputTypesAfterOriginalResolver` to `false`, what means that the directive don't need the value to be validated and, if it fails, we don't want to spend time applying the resolver.

In another words, the validation function of a directive implementation that overloads the  `applyValidationToOutputTypesAfterOriginalResolver` to `false` should not use the value argument. I agree that this dependency  isn't as clear as I would like, but I couldn't find another way to do it